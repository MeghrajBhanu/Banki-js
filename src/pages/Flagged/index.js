import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { get_all_flag_accounts } from "../../redux/actions/pan_actions";
import BankAccountsMap from "../../components/BankAccountsMap";
import classes from "../Landing/index.module.css";
import SharedHeader from "../../components/SharedHeader";

/**
 * Flagged Page return All the flagged Accounts
 * @returns {React.ReactElement}
 */
const Flagged = () => {
  const user = useSelector((state) => state.auth.user);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(get_all_flag_accounts(user?.pancard?user["pancard"]:"LTGWS1234L"));
  }, []);
  
  const data = useSelector((state) => state.pan.data);
  const [page, setPage] = useState(1);
  const selectPageHandler = (page_i) => {
    if (
      page_i >= 1 &&
      page_i <= [...Array(Math.ceil(data.length / 10))].length &&
      page_i !== page
    )
      setPage(page_i);
  };

  let mapped = data?.slice(page * 10 - 10, page * 10).map((item) => {
    return (
      <BankAccountsMap
        key={item._id}
        id={item._id}
        name={item.name}
        email={item.email}
        bankName={item.bankName}
        pancard={item.pancard}
        account_type={item.AccountType}
        fd={item.FixedDeposits}
        balance={item.Balance}
        flagged={true}
      ></BankAccountsMap>
    );
  });
  return (
    <>
    <div className="mb-4">
            <SharedHeader
              name={user?.name}
              pancard={user?.pancard}
              flagged={"flagged"}
            />
          </div>
      {data?.length > 0 ? (
        <>
          
          <div className={classes["cards"]}>{mapped}</div>
          {data.length > 10 && (
            <div
              className={classes["pagination"]}
              style={{
                padding: "10px",
                margin: "15px 0",
                display: "flex",
                justifyContent: "center",
                alignItems: "center",
              }}
            >
              <span
                onClick={() => selectPageHandler(page - 1)}
                className={page > 1 ? "" : classes["disable_button"]}
              >
                ◀
              </span>

              {[...Array(Math.ceil(data.length / 10))].map((_, i) => {
                return (
                  <span
                    key={i}
                    className={page === i + 1 ? classes["selected"] : ""}
                    onClick={() => selectPageHandler(i + 1)}
                  >
                    {i + 1}
                  </span>
                );
              })}

              <span
                onClick={() => selectPageHandler(page + 1)}
                className={
                  page < data.length / 10 ? "" : classes["disable_button"]
                }
              >
                ▶
              </span>
            </div>
          )}
        </>
      ) : (
        <>
          <h5 className="text-center mb-5 mt-5">OOPS!!</h5>
          <h6 className="text-center ">There are no flagged accounts </h6>
        </>
      )}
    </>
  );
};
export default Flagged;
