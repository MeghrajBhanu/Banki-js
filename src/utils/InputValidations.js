export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}