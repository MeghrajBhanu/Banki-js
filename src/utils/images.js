import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;