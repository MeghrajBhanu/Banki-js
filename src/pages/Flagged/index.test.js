import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import Flagged from ".";

describe("Flagged", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<Flagged />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render the  shared header anf filter button", () => {
    renderComponent(<Flagged />);
    const welcome = screen.findByText(/welcome/i);

    waitFor(() => {
      expect(welcome).toBeInTheDocument();
    });
  });
  test("data in form ofcards should be rendered", () => {
    renderComponent(<Flagged />);

    const AccountType = screen.findByText(/salary/i);
    const bankName = screen.findByText(/Bank3/i);
    const view = screen.findByRole("button", {
      name: /UnFlag/i,
    });
    waitFor(() => {
      expect(AccountType).not.toBeInTheDocument();
      expect(bankName).not.toBeInTheDocument();
      expect(view).not.toBeInTheDocument();
    });
  });
  test("redux store should recieve data", () => {
    let { store } = renderComponent(<Flagged />);
    expect(store.getState().pan.data.length === 0);
  });
  test("shouldhandle case were no flagged accounts are found", () => {
    renderComponent(<Flagged />);
    screen.logTestingPlaygroundURL();
    const oops = screen.findByRole("heading", {
      name: /oops!!/i,
    });
    const noaccounts = screen.findByRole("heading", {
      name: /there are no flagged accounts/i,
    });
    waitFor(() => {
      expect(noaccounts).toBeInTheDocument();
      expect(oops).toBeInTheDocument();
    });
  });
});
