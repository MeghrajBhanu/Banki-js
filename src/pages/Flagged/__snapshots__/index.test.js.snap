// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flagged snapshot test 1`] = `
Array [
  <div
    className="mb-4"
  >
    <section
      className="shared_header"
    >
      <div
        className="overlay"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col"
              data-wow-delay="0.6s"
              style={
                Object {
                  "animationDelay": "0.6s",
                  "animationName": "fadeInUp",
                  "visibility": "visible",
                }
              }
            >
              <div
                className="fadeInUp"
              >
                <h2
                  className="text-center mb-4"
                >
                  Welcome 
                  user
                   !
                </h2>
                <p
                  className="text-center mt-3"
                >
                  Here are your 
                  flagged
                   bank accounts linked with 
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  <h5
    className="text-center mb-5 mt-5"
  >
    OOPS!!
  </h5>,
  <h6
    className="text-center "
  >
    There are no flagged accounts 
  </h6>,
]
`;

exports[`snapshot test 1`] = `
Array [
  <div
    className="mb-4"
  >
    <section
      className="shared_header"
    >
      <div
        className="overlay"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col"
              data-wow-delay="0.6s"
              style={
                Object {
                  "animationDelay": "0.6s",
                  "animationName": "fadeInUp",
                  "visibility": "visible",
                }
              }
            >
              <div
                className="fadeInUp"
              >
                <h2
                  className="text-center mb-4"
                >
                  Welcome 
                  user
                   !
                </h2>
                <p
                  className="text-center mt-3"
                >
                  Here are your 
                  flagged
                   bank accounts linked with 
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  <h5
    className="text-center mb-5 mt-5"
  >
    OOPS!!
  </h5>,
  <h6
    className="text-center "
  >
    There are no flagged accounts 
  </h6>,
]
`;
